distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso

```diff
@@ -310,12 +310,32 @@ type Not_Invokable
 
        Arguments:
        - target: The called object.
-    Error target
+       - cause: Additional information about what may have caused the error.
+    Error target cause
 
     ## PRIVATE
        Convert the Not_Invokable error to a human-readable format.
     to_display_text : Text
-    to_display_text self = "Type error: expected a function, but got "+self.target.to_display_text+"."
+    to_display_text self =
+        suffix = case self.cause of
+            Nothing -> ""
+            _ -> " Caused by: "+self.cause.to_display_text
+        "Type error: expected a function, but got "+self.target.to_display_text+"."+suffix
+
+@Builtin_Type
+type No_Such_Argument
+    ## PRIVATE
+       Indicates that an argument was passed by name, but the function being
+       called did not take any argument that matched that name.
+
+       Arguments:
+       - argument_name: The name of the argument that was not found.
+    Error argument_name
```

Comment:
Remove `## PRIVATE` use:
```suggestion
    private Error argument_name
```
possibly remove the documentation comment altogether.

Reply:
This is against the convention that we are using for all the errors.

I think changes in this PR should be consistent with the rest of the codebase - if we want to change the convention let's discuss and consider rewriting the whole `Common.enso` file.

Reply:
We don't want the constructor to be `private`, this breaks the very simple check I've implemented like `payload.cause == (No_Such_Argument.Error "foo")`, at least outside of Base.

The `PRIVATE` is to avoid cluttering Component Browser only, I guess at some point we may remove that tag from errors.